waspc/src/Wasp/Analyzer/Evaluator/EvaluationError.hs

```diff
@@ -1,33 +1,47 @@
+{-# LANGUAGE LambdaCase #-}
+
 module Wasp.Analyzer.Evaluator.EvaluationError
   ( EvaluationError (..),
-    EvaluationErrorContext (..),
+    EvalErrorCtx (..),
     EvaluationParseError (..),
+    EvaluationError' (..),
+    mkEvaluationError,
+    getErrorMessageAndCtx,
   )
 where
 
+import Control.Arrow (first)
+import Data.List (intercalate)
 import qualified Text.Parsec
+import Wasp.Analyzer.Parser.Ctx (Ctx, WithCtx (..))
 import Wasp.Analyzer.Type (Type)
+import Wasp.Util (concatPrefixAndText, indent)
+
+newtype EvaluationError = EvaluationError (WithCtx EvaluationError')
```

Comment:
Here for the first time I used WithCtx to put Ctx in the error, and not just AST. In the later PR I also do this for TypeError.

The only "ugly" thing is that I had a bit of hard time naming `data EvaluationError'`. I could name them `EvaluationErrorWithCtx` and `EvaluationError`, but the thing is that in the code we mostly use the first one and don't care about the second one much, so having such a long and overly detailed name sounded impractical. Therefore I decided to just call it `EvaluationError` and `EvaluationError'`. One other way to do it could be `EvaluationError` and `EvaluationErrorReason`/`EvalautionErrorCause`, but again I didn't want to over complicate.